### PR TITLE
VB-650 Change RequestsHandler API to support metadata state sequence number.

### PR DESCRIPTION
**Pull request by yuliasherman**
This commit contains minimal changes required to add a sequence number parameter to RequestsHandler execute() function API.

### CODE REVIEW

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 1]
```diff
@@ -3089,9 +3094,13 @@ namespace bftEngine
 
 
 				uint32_t actualReplyLength = 0;
-
-				int error = userRequestsHandler->execute(clientId, req.isReadOnly(), req.requestLength(), req.requestBuf(),
-															maxReplyMessageSize - sizeof(ClientReplyMsgHeader), replyBuffer, actualReplyLength);
+				// Yulia TBD: sequenceNum should come from corresponding State
+				// Metadata object.
+				int error = userRequestsHandler->execute(
```
Comment (guyg8) [inline bftengine/src/bftengine/ReplicaImp.cpp]: Instead of 0. Pass lastExecutedSeqNum + 1

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 2]
```diff
@@ -3034,7 +3034,12 @@ namespace bftEngine
 
 			if (!supportDirectProofs)
 			{
-				error = userRequestsHandler->execute(clientId, true, request->requestLength(), request->requestBuf(), reply.maxReplyLength(), reply.replyBuf(), actualReplyLength);
+				// Yulia TBD: sequenceNum should come from corresponding State
+				// Metadata object.
+				error = userRequestsHandler->execute(
+				    clientId, 0, true, request->requestLength(),
+				    request->requestBuf(), reply.maxReplyLength(),
```
Comment (guyg8) [inline bftengine/src/bftengine/ReplicaImp.cpp]: Instead of 0. Pass lastExecutedSeqNum

Comment (yuliasherman) [reply bftengine/src/bftengine/ReplicaImp.cpp]: Done.

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 1 above)
Comment (yuliasherman) [reply bftengine/src/bftengine/ReplicaImp.cpp]: Done.

**Pull request merged**